src/bin/jcli_app/utils/output_format.rs

```diff
@@ -0,0 +1,83 @@
+use gtmpl::Value as GtmplValue;
+use jcli_app::utils::error::CustomErrorFiller;
+use serde::Serialize;
+use serde_json::{Map as JsonMap, Number as JsonNumber, Value as JsonValue};
+use std::fmt::{self, Display, Formatter};
+use structopt::StructOpt;
+
+#[derive(StructOpt)]
+pub struct OutputFormat {
+    /// Format of output data. Possible values: json, yaml.
+    /// Any other value is treated as a custom format using values from output data structure.
+    /// Syntax is Go text template: https://golang.org/pkg/text/template/.
+    #[structopt(long = "output-format", default_value = "yaml")]
+    format: String,
+}
+
+custom_error! { pub Error
+    YamlFormattingFailed { source: serde_yaml::Error } = "failed to format output as YAML",
+    JsonFormattingFailed { source: serde_json::Error } = "failed to format output as JSON",
+    JsonValueCreateFailed { source: serde_json::Error, filler: CustomErrorFiller }
+        = "failed to create intermediate formatting data",
+    CustomFormattingFailed { source: GtmplError } = "failed to format output as custom format",
+}
+
+#[derive(Debug)]
+pub struct GtmplError(String);
+
+impl Display for GtmplError {
+    fn fmt(&self, f: &mut Formatter) -> Result<(), fmt::Error> {
+        self.0.fmt(f)
+    }
+}
+
+impl std::error::Error for GtmplError {}
+
+impl OutputFormat {
+    pub fn format(&self, data: impl Serialize) -> Result<String, Error> {
+        Ok(match self.format.trim().to_ascii_lowercase().as_str() {
```

Comment:
The `.trim().to_ascii_lowercase()` normalization should be done on construction of `OutputFormat`.

Reply:
The normalization is not a valid step for custom format definiton, it can be used only for `json` and `yaml` flag detection

Reply:
The constructor could check the normalized value for these two matches and set the member value as is if it's something else.

Reply:
This is the idea of making `OutputFormat` an enum coming back, isn't it?

Reply:
No, just normalizing the input once in the proper place.